src/tpt_reports/report_generator.py

```diff
@@ -0,0 +1,414 @@
+    doc.addPageTemplates(
+        [
+            PageTemplate(id="title_page", frames=title_frame, onPage=title_page),
+            PageTemplate(id="content_page", frames=content_frame, onPage=content_page),
+        ]
+    )
+    # Generate the list to fill all the dynamic content streamed through the report
+    Story = []
+    paragraph_style = ParagraphStyle
+    image_text = paragraph_style(
+        name="image_text",
+        fontName="Franklin_Gothic_Medium_Regular",
+        fontSize=8,
+        leading=16,
+        alignment=1,
+        spaceAfter=10,
+        spaceBefore=10,
+        textColor=HexColor("#003e67"),
+    )
+    cover_heading = paragraph_style(
+        fontName="Franklin_Gothic_Medium_Regular",
+        name="Heading1",
+        fontSize=28,
+        leading=48,
+        spaceBefore=10,
+        textColor=HexColor("#003e67"),
+    )
+    h2 = paragraph_style(
+        name="Heading2",
+        fontName="Franklin_Gothic_Medium_Regular",
+        fontSize=14,
+        leading=10,
+        textColor=HexColor("#003e67"),
+        spaceAfter=12,
+    )
+    body = paragraph_style(
+        name="body",
+        leading=14,
+        fontName="Franklin_Gothic_Book",
+        fontSize=12,
+    )
+    table = paragraph_style(
+        name="table",
+        fontName="Franklin_Gothic_Medium_Regular",
+        fontSize=18,
+        alignment=1,
+        spaceAfter=14,
+    )
+    table_header = paragraph_style(
+        name="table_header",
+        fontName="Franklin_Gothic_Medium_Regular",
+        fontSize=12,
+        leading=20,
+        alignment=1,
+        spaceAfter=30,
+        textColor=HexColor("#FFFFFF"),
+    )
+
+    # Create dynamic content; repeated and random elements used in the report
+    point12_spacer = ConditionalSpacer(1, 12)
+    point24_spacer = ConditionalSpacer(1, 24)
+
+    # Appends sequentially with the frames created above i.e. title_page then content_page
+    Story.append(get_image(f"{BASE_DIR}/assets/TitlePage.png", width=9 * inch))
+    Story.append(point24_spacer)
+    Story.append(
+        Paragraph(
+            f"""Technical Payload Test (TPT) Report - {tpt_info["election_name"]}""",
+            cover_heading,
+        )
+    )
+    Story.append(point24_spacer)
+    Story.append(NextPageTemplate("content_page"))
+    Story.append(PageBreak())
+    Story.append(Paragraph("TPT Report", table))
+    Story.append(
+        format_table(
+            pd.DataFrame.from_dict(
+                {
+                    "Report Date": TODAYS_DATE,
+                    "Stakeholder Name": [tpt_info["election_name"]],
+                    "Domain Tested": [tpt_info["domain_tested"]],
+                    "Assessment ID": [tpt_info["servicenow_id"]],
+                }
+            ),
+            table_header,
+            [1.2 * inch, 2.2 * inch, 1.6 * inch, 1.3 * inch],
+            [None, body, body, body],
+        )
+    )
+    Story.append(Spacer(1, 30))
```

Comment:
This spacer is also used more than once, so might as well replace it with a variable like you did with `point12_spacer` and `point24_spacer`.

Reply:
Resolved in [40827a5](https://github.com/cisagov/tpt-reports/pull/15/commits/40827a53d917b09116ade510dcd089be3c03a0bb)